src/fetch.ts

```diff
@@ -46,11 +49,15 @@ export function createFetch(globalOptions: CreateFetchOptions = {}): $Fetch {
         context.error.name === "AbortError" &&
         !context.options.timeout) ||
       false;
+
     // Retry
     if (context.options.retry !== false && !isAbort) {
       let retries;
       if (typeof context.options.retry === "number") {
         retries = context.options.retry;
+        if (retries > MAX_RETRIES) {
```

Comment:
What about introducing new `maxRetries` option to allow setting the default?

Reply:
I wonder if this defeats the purpose of this PR. My understanding was to include a sensible default for maximum retries. Instead of a `maxRetries` option, `retries` basically does the same, right?
If you dislike the idea of an internal maximum, which I completely get, please close this PR. 🙂